nushell_completions.go

```diff
@@ -0,0 +1,175 @@
+            spans: ($spans | drop | append ($split | first) | append ($split | last))
+          }
+        } else {
+          {
+            last_span: '',
+            spans: ($spans | drop | append ($split | first) | append '')
+          }
+        }
+      } else {
+        $params
+      }
+
+      let last_span = $params.last_span
+      let spans = $params.spans
+
+      # Drop the last param so we can fuzzy search on it
+      let spans = if $fuzzy {
+        $spans | drop
+      } else {
+        $spans
+      }
+
+      # skip the first entry in the span (the command) and join the rest of the span to create __complete args
+      let cmd_args = ($spans | skip 1 | str join ' ')
+
+      # If the last span entry was empty add "" to the end of the command args
+      let cmd_args = if ($last_span | is-empty) or $fuzzy {
+        $'($cmd_args) ""'
+      } else {
+        $cmd_args
+      }
+
+      # The full command to be executed with active help disable (Nushell does not support active help)
+      let full_cmd = $'COBRA_ACTIVE_HELP=0 ($cmd) __complete ($cmd_args)'
+
+      # Since nushell doesn't have anything like eval, execute in a subshell
+      let result = (do -i { nu -c $"'($full_cmd)'" } | complete)
+
+      # Create a record with all completion related info.
+      # directive and directive_str are for posterity
+      let stdout_lines = ($result.stdout | lines)
+      let directive = ($stdout_lines | last | str trim | str replace ":" "" | into int)
+      let completions = ($stdout_lines | drop | parse -r '([\w\-\.:\+\=\/]*)\t?(.*)' | rename value description)
+      let completions = if $fuzzy {
+        $completions | where $it.value =~ $last_span
+
+      } else {
+        ($completions | where {|it| $it.value | str starts-with $last_span })
+      }
+
+      {
+        directive: $directive,
+        completions: $completions
+      }
+    }
+
+    let result = (exec_complete $spans)
+    let result = if (not ($last_span | is-empty)) and ($result.completions | is-empty) {
+      exec_complete --fuzzy $spans
+    } else {
+      $result
+    }
+
+    let directive = $result.directive
+    let completions = $result.completions
+
+    # Add space at the end of each completion
+    let completions = if $directive != $ShellCompDirectiveNoSpace {
+      $completions | each {|it| {value: $"($it.value) ", description: $it.description}}
+    } else {
+      $completions
+    }
+
+    # Cobra returns a list of completions that are supported with this directive
+    # There is no way to currently support this in a nushell external completer
+    let completions = if $directive == $ShellCompDirectiveFilterFileExt {
+      []
+    } else {
+      $completions
+    }
+
+    let return_val = if $last_span =~ '=' {
+      # if the completion is of the form -n= return flag as part of the completion so that it doesn't get replaced
+      $completions | each {|it| $"($last_span | split row '=' | first)=($it.value)" }
```

Comment:
It is valid for a program to request file completion for a flag value.  For example:
```
kubectl apply -f <tab>
```
This currently works.
However, if we use the `=` for it does not:
```
kubectl apply -f=<tab>
```
I'm not sure if you will be to handle that case or not.  If not, we'll just have to live with it.  It's not critical.